src/test/java/calculator/ArithmeticOperator.java

```diff
@@ -0,0 +1,39 @@
+package calculator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.function.BiFunction;
+
+public enum ArithmeticOperator {
+
+    PLUS((a, b) -> a + b, "+"),
+    MINUS((a, b) -> a - b, "-"),
+    MULT((a, b) -> a * b, "*"),
+    DIV((a, b) -> a / b, "/");
+
+    public static final int REQUIRED_NUM_OPERANDS = 2;
+
+    private BiFunction<Long, Long, Long> biFunction;
+    private String mark;
+
+    ArithmeticOperator(BiFunction<Long, Long, Long> f, String mark) {
+        this.biFunction = f;
+        this.mark = mark;
+    }
+
+    public static ArithmeticOperator of(String token) {
+        return Arrays.stream(values())
+                .filter(ao -> ao.mark.equals(token))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("not an operator"));
+    }
+
+    public long eval(ArrayList<Long> operands) {
```

Comment:
```suggestion
    public long eval(List<Long> operands) {
```
구체적은 컬렉션 보다는 List 로 선언하여 유연하게 사용하는건 어떨까요 ?
https://stackoverflow.com/questions/2279030/type-list-vs-type-arraylist-in-java

Reply:
이 코드에선 사실 적절한 예는 아니긴 합니다만, 구체적인 컬렉션을 나타내야 사용된 자료구조가 어떤 것이니 어떻게 접근해야 할지 표현가능할 것 같습니다.
예를 들어 random access가 필요한데 list 타입으로 받은게 알고봤더니 linked list 인 경우라면요.

Reply:
좋은 예시를 들어주셔서 감사합니다 👍 
저는 개인적으론 확장성 있는 기능 구현을 위해 구체적인 컬렉션을 선언하지 않는 편입니다. 
예를 들면 HashSet으로 구현 되어 있는 컬렉션을 TreeSet 으로 변경하는 하는 경우 변화에 유연하게 대처할 수 있으니깐요 🙏 
아래 문서를 참고해보시는것도 좋을것 같아요 ! 
https://velog.io/@ybg7955/Effective-Java-3E-9%EC%9E%A5-%EC%9D%BC%EB%B0%98%EC%A0%81%EC%9D%B8-%ED%94%84%EB%A1%9C%EA%B7%B8%EB%9E%98%EB%B0%8D-%EC%9B%90%EC%B9%99#%EC%95%84%EC%9D%B4%ED%85%9C-64-%EA%B0%9D%EC%B2%B4%EB%8A%94-%EC%9D%B8%ED%84%B0%ED%8E%98%EC%9D%B4%EC%8A%A4%EB%A5%BC-%EC%82%AC%EC%9A%A9%ED%95%B4-%EC%B0%B8%EC%A1%B0%ED%95%98%EB%9D%BC